smartlearner/optimizers/adagrad.py

```diff
@@ -0,0 +1,55 @@
+from collections import OrderedDict
+import theano.tensor as T
+
+from . import SGD
+from ..utils import sharedX
+
+
+class ADAGRAD(SGD):
+    """
+    Implements the ADAGRAD optimizer.
+
+    Reference
+    ---------
+    Duchi, J., Hazan, E., & Singer, Y. (2010).
+    Adaptive subgradient methods for online learning and stochastic optimization.
+    Journal of Machine Learning
```

Comment:
Should be : 

    .. Duchi, J., Hazan, E., & Singer, Y. (2010).
       Adaptive subgradient methods for online learning and stochastic optimization.
       Journal of Machine Learning